MSFT_EXOAddressList: You can't use customized filters and precanned filters at the same time

<!--
    Your feedback and support is greatly appreciated, thanks for contributing!

    ISSUE TITLE:
    Please prefix the issue title with the resource name, e.g.
    'ResourceName: Short description of my issue'

    ISSUE DESCRIPTION (this template):
    Please provide information regarding your issue under each header below.
    Write N/A under any headers that do not apply to your issue, or if the
    information is not available.

    NOTE! Sensitive information should be obfuscated.

    PLEASE KEEP THE HEADERS.

    You may remove this comment block, and the other comment blocks,
    but please keep the headers.
-->
#### Details of the scenario you tried and the problem that is occurring
Using the AddressList DscResource, it produces an error due to the use of a combination of filters. However, the provided input is valid. The issue seems to reside in the 'empty' key-value pairs of either NewAddressListParams or SetAddressListParams.

#### Verbose logs showing the problem
```
VERBOSE: [hostname]: LCM:  [ Start  Resource ]  [[EXOAddressList]AllEquipment]
VERBOSE: [hostname]: LCM:  [ Start  Test     ]  [[EXOAddressList]AllEquipment]
VERBOSE: [hostname]:                            [[EXOAddressList]AllEquipment] Testing Address List configuration for All Equipment
VERBOSE: [hostname]:                            [[EXOAddressList]AllEquipment] Getting configuration of AddressList for All Equipment
VERBOSE: [hostname]:                            [[EXOAddressList]AllEquipment] Found AddressList All Equipment
VERBOSE: [hostname]:                            [[EXOAddressList]AllEquipment] Get-TargetResource Result: 
 ApplicationId=***; CertificatePassword=***; CertificatePath=***; 
CertificateThumbprint=***; ConditionalCompany=; ConditionalCustomAttribute1=; 
ConditionalCustomAttribute10=; ConditionalCustomAttribute11=; ConditionalCustomAttribute12=; 
ConditionalCustomAttribute13=; ConditionalCustomAttribute14=; ConditionalCustomAttribute15=; 
ConditionalCustomAttribute2=; ConditionalCustomAttribute3=; ConditionalCustomAttribute4=; ConditionalCustomAttribute5=;
 ConditionalCustomAttribute6=; ConditionalCustomAttribute7=; ConditionalCustomAttribute8=; 
ConditionalCustomAttribute9=; ConditionalDepartment=; ConditionalStateOrProvince=; Credential=$null; DisplayName=All 
Equipment; Ensure=Present; IncludedRecipients=(); Name=All Equipment; RecipientFilter=((Alias -ne $null) -and 
(RecipientTypeDetails -eq 'EquipmentMailbox')); TenantId=***

VERBOSE: [hostname]:                            [[EXOAddressList]AllEquipment] Current Values: ApplicationId=***; 
CertificatePassword=***; CertificatePath=***; CertificateThumbprint=***; 
ConditionalCompany=; ConditionalCustomAttribute1=; ConditionalCustomAttribute10=; ConditionalCustomAttribute11=; 
ConditionalCustomAttribute12=; ConditionalCustomAttribute13=; ConditionalCustomAttribute14=; 
ConditionalCustomAttribute15=; ConditionalCustomAttribute2=; ConditionalCustomAttribute3=; 
ConditionalCustomAttribute4=; ConditionalCustomAttribute5=; ConditionalCustomAttribute6=; ConditionalCustomAttribute7=;
 ConditionalCustomAttribute8=; ConditionalCustomAttribute9=; ConditionalDepartment=; ConditionalStateOrProvince=; 
Credential=$null; DisplayName=All Equipment; Ensure=Present; IncludedRecipients=(); Name=All Equipment; 
RecipientFilter=((Alias -ne $null) -and (RecipientTypeDetails -eq 'EquipmentMailbox')); TenantId=***

VERBOSE: [hostname]:                            [[EXOAddressList]AllEquipment] Target Values: ApplicationId=***; 
CertificateThumbprint=***; DisplayName=All Equipment; Ensure=Present; Name=All 
Equipment; RecipientFilter=((Alias -ne $null) -and (RecipientTypeDetails -eq "EquipmentMailbox")); TenantId=***; 
Verbose=True

VERBOSE: [hostname]:                            [[EXOAddressList]AllEquipment] Test-TargetResource returned False
VERBOSE: [hostname]: LCM:  [ End    Test     ]  [[EXOAddressList]AllEquipment]  in 0.6110 seconds.
VERBOSE: [hostname]: LCM:  [ Start  Set      ]  [[EXOAddressList]AllEquipment]
VERBOSE: [hostname]:                            [[EXOAddressList]AllEquipment] Setting Address List configuration for All Equipment
VERBOSE: [hostname]:                            [[EXOAddressList]AllEquipment] Getting configuration of AddressList for All Equipment
VERBOSE: [hostname]:                            [[EXOAddressList]AllEquipment] Found AddressList All Equipment
VERBOSE: [hostname]:                            [[EXOAddressList]AllEquipment] Get-TargetResource Result: 
 ApplicationId=***; CertificatePassword=***; CertificatePath=***; 
CertificateThumbprint=***; ConditionalCompany=; ConditionalCustomAttribute1=; 
ConditionalCustomAttribute10=; ConditionalCustomAttribute11=; ConditionalCustomAttribute12=; 
ConditionalCustomAttribute13=; ConditionalCustomAttribute14=; ConditionalCustomAttribute15=; 
ConditionalCustomAttribute2=; ConditionalCustomAttribute3=; ConditionalCustomAttribute4=; ConditionalCustomAttribute5=;
 ConditionalCustomAttribute6=; ConditionalCustomAttribute7=; ConditionalCustomAttribute8=; 
ConditionalCustomAttribute9=; ConditionalDepartment=; ConditionalStateOrProvince=; Credential=$null; DisplayName=All 
Equipment; Ensure=Present; IncludedRecipients=(); Name=All Equipment; RecipientFilter=((Alias -ne $null) -and 
(RecipientTypeDetails -eq 'EquipmentMailbox')); TenantId=***

VERBOSE: [hostname]:                            [[EXOAddressList]AllEquipment] Address List 'All Equipment' already exists. Updating settings
VERBOSE: [hostname]:                            [[EXOAddressList]AllEquipment] Setting Address List 'All Equipment' 
with values: ConditionalCompany=$null; ConditionalCustomAttribute1=$null; ConditionalCustomAttribute10=$null; 
ConditionalCustomAttribute11=$null; ConditionalCustomAttribute12=$null; ConditionalCustomAttribute13=$null; 
ConditionalCustomAttribute14=$null; ConditionalCustomAttribute15=$null; ConditionalCustomAttribute2=$null; 
ConditionalCustomAttribute3=$null; ConditionalCustomAttribute4=$null; ConditionalCustomAttribute5=$null; 
ConditionalCustomAttribute6=$null; ConditionalCustomAttribute7=$null; ConditionalCustomAttribute8=$null; 
ConditionalCustomAttribute9=$null; ConditionalDepartment=$null; ConditionalStateOrProvince=$null; Confirm=False; 
DisplayName=All Equipment; Identity=All Equipment; IncludedRecipients=$null; Name=All Equipment; 
RecipientFilter=((Alias -ne $null) -and (RecipientTypeDetails -eq "EquipmentMailbox"))

##[error]Failed to start the DSC configuration. Error: You can't use customized filters and precanned filters at the same time.
```

#### Suggested solution to the issue
According to the [Docs](https://docs.microsoft.com/en-us/powershell/exchange/recipient-filters?view=exchange-ps#precanned-filters), the following are considered precanned filters:

- IncludedRecipients
- ConditionalCompany
- ConditionalDepartment
- ConditionalStateOrProvince
- ConditionalCustomAttribute1 to ConditionalCustomAttribute15.

Suggested to avoid this all together, and update the DscResource to remove all empty KeyValue pairs. Any error then still produced might be due to the provided input, combining custom filters with precanned filters, but that's an input-error. However, I don't know if this would cause conflicts (on update) when moving from precanned to custom filters or viceversa.

```powershell
($NewAddressListParams.GetEnumerator() | Where-Object { $_.Value -eq $null }) | Foreach-Object { $NewAddressListParams.Remove($_.Name) }

($SetAddressListParams.GetEnumerator() | Where-Object { $_.Value -eq $null }) | Foreach-Object { $SetAddressListParams.Remove($_.Name) }
```

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
Node localhost {
	EXOAddressList 'AllEquipment' {
		# Authentication details
		#Credential            = $CredsGlobalAdmin
		TenantId               = $TenantId
		ApplicationId          = $ApplicationId
		CertificateThumbprint  = $CertificateThumbprint
 
		# Settings             
		Ensure                 = "Present"
		Name                   = "All Equipment"
		DisplayName            = "All Equipment"
		RecipientFilter        = '((Alias -ne $null) -and (RecipientTypeDetails -eq "EquipmentMailbox"))'
	}
}
```

#### The operating system the target node is running
<!--
    Please provide as much as possible about the target node, for example
    edition, version, build and language.
    On OS with WMF 5.1 the following command can help get this information.

    Get-ComputerInfo -Property @(
        'OsName',
        'OsOperatingSystemSKU',
        'OSArchitecture',
        'WindowsVersion',
        'WindowsBuildLabEx',
        'OsLanguage',
        'OsMuiLanguages')
-->
```
OsName               : Microsoft Windows Server 2019 Datacenter
OsOperatingSystemSKU : DatacenterServerEdition
OsArchitecture       : 64-bit
WindowsVersion       : 1809
WindowsBuildLabEx    : 17763.1.amd64fre.rs5_release.180914-1434
OsLanguage           : en-US
OsMuiLanguages       : {en-US}
```

#### Version of the DSC module that was used ('dev' if using current dev branch)
1.22.713.1

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,9 @@
 
 # UNRELEASED
 
+* EXOAddressList
+  * Ignore precanned filters if recipient filter is used. Precanned filters and recipient filter cannot be used at the same time.
+    FIXES [#2194](https://github.com/microsoft/Microsoft365DSC/issues/2194)
 * EXOSafeLinksPolicy
   * Add Suport for EnableSafeLinksForEmail and DisableUrlRewrite
 * EXOInboundConnector

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_EXOAddressList/MSFT_EXOAddressList.psm1
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_EXOAddressList/MSFT_EXOAddressList.psm1
@@ -395,6 +395,7 @@ function Set-TargetResource
 
     if ($RecipientFilter)
     {
+        Write-Verbose -Message "You can't use RecipientFilter and precanned filters at the same time. All precanned filters will be ignored."
         $NewAddressListParams = @{
             Name            = $Name
             RecipientFilter = $RecipientFilter
@@ -429,31 +430,44 @@ function Set-TargetResource
         }
     }
 
-    $SetAddressListParams = @{
-        Identity                     = $Name
-        Name                         = $Name
-        ConditionalCompany           = $ConditionalCompany
-        ConditionalCustomAttribute1  = $ConditionalCustomAttribute1
-        ConditionalCustomAttribute10 = $ConditionalCustomAttribute10
-        ConditionalCustomAttribute11 = $ConditionalCustomAttribute11
-        ConditionalCustomAttribute12 = $ConditionalCustomAttribute12
-        ConditionalCustomAttribute13 = $ConditionalCustomAttribute13
-        ConditionalCustomAttribute14 = $ConditionalCustomAttribute14
-        ConditionalCustomAttribute15 = $ConditionalCustomAttribute15
-        ConditionalCustomAttribute2  = $ConditionalCustomAttribute2
-        ConditionalCustomAttribute3  = $ConditionalCustomAttribute3
-        ConditionalCustomAttribute4  = $ConditionalCustomAttribute4
-        ConditionalCustomAttribute5  = $ConditionalCustomAttribute5
-        ConditionalCustomAttribute6  = $ConditionalCustomAttribute6
-        ConditionalCustomAttribute7  = $ConditionalCustomAttribute7
-        ConditionalCustomAttribute8  = $ConditionalCustomAttribute8
-        ConditionalCustomAttribute9  = $ConditionalCustomAttribute9
-        ConditionalDepartment        = $ConditionalDepartment
-        ConditionalStateOrProvince   = $ConditionalStateOrProvince
-        DisplayName                  = $DisplayName
-        IncludedRecipients           = $IncludedRecipients
-        RecipientFilter              = $RecipientFilter
-        Confirm                      = $false
+    if ($RecipientFilter)
+    {
+        Write-Verbose -Message "You can't use RecipientFilter and precanned filters at the same time. All precanned filters will be ignored."
+        $SetAddressListParams = @{
+            Identity        = $Name
+            Name            = $Name
+            RecipientFilter = $RecipientFilter
+            Confirm         = $false
+        }
+    }
+    else
+    {
+        $SetAddressListParams = @{
+            Identity                     = $Name
+            Name                         = $Name
+            ConditionalCompany           = $ConditionalCompany
+            ConditionalCustomAttribute1  = $ConditionalCustomAttribute1
+            ConditionalCustomAttribute10 = $ConditionalCustomAttribute10
+            ConditionalCustomAttribute11 = $ConditionalCustomAttribute11
+            ConditionalCustomAttribute12 = $ConditionalCustomAttribute12
+            ConditionalCustomAttribute13 = $ConditionalCustomAttribute13
+            ConditionalCustomAttribute14 = $ConditionalCustomAttribute14
+            ConditionalCustomAttribute15 = $ConditionalCustomAttribute15
+            ConditionalCustomAttribute2  = $ConditionalCustomAttribute2
+            ConditionalCustomAttribute3  = $ConditionalCustomAttribute3
+            ConditionalCustomAttribute4  = $ConditionalCustomAttribute4
+            ConditionalCustomAttribute5  = $ConditionalCustomAttribute5
+            ConditionalCustomAttribute6  = $ConditionalCustomAttribute6
+            ConditionalCustomAttribute7  = $ConditionalCustomAttribute7
+            ConditionalCustomAttribute8  = $ConditionalCustomAttribute8
+            ConditionalCustomAttribute9  = $ConditionalCustomAttribute9
+            ConditionalDepartment        = $ConditionalDepartment
+            ConditionalStateOrProvince   = $ConditionalStateOrProvince
+            DisplayName                  = $DisplayName
+            IncludedRecipients           = $IncludedRecipients
+            RecipientFilter              = $RecipientFilter
+            Confirm                      = $false
+        }
     }
 
     #Address List doesn't exist but it should

